src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Queries/GetApprenticeship/GetApprenticeshipConfirmationStatusTests.cs

```diff
@@ -29,6 +29,71 @@ public GetApprenticeshipConfirmationStatusTests()
             _query = new GetApprenticeshipQuery(_apprenticeshipID);
         }
 
+        [Test]
+        public async Task WhenEmailIsNull_ThenEmailAddressConfirmedByApprenticeShouldBeFalse()
+        {
+            // Arrange
+            Setup(null, null, null);
+
+            // Act
+            var response = await _sut.Handle(_query, new CancellationToken());
+
+            //Assert
+            Assert.AreEqual(response.EmailAddressConfirmedByApprentice, false);
+        }
+
+        [Test]
+        public async Task WhenEmailIsNotNull_ThenEmailAddressConfirmedByApprenticeShouldBeTrue()
+        {
+            // Arrange
+            Setup("test@test.com", null, null);
+
+            // Act
+            var response = await _sut.Handle(_query, new CancellationToken());
+
+            //Assert
+            Assert.AreEqual(response.EmailAddressConfirmedByApprentice, true);
+        }
+
+        [Test]
+        public async Task WhenCohortIsApprovedAfter9Sept21_ThenEmailShouldBePresentShouldBeTrue()
+        {
+            // Arrange
+            Setup(null, null, null);
+
+            // Act
+            var response = await _sut.Handle(_query, new CancellationToken());
+
+            //Assert
+            Assert.AreEqual(response.EmailShouldBePresent, true);
+        }
+
+        [Test]
+        public async Task WhenCohortIsApprovedAfter9Sept21ButIsAContinuationApprenticeship_ThenEmailShouldBePresentShouldBeTrue()
+        {
+            // Arrange
+            Setup(null, null, null, null, 10890);
+
+            // Act
+            var response = await _sut.Handle(_query, new CancellationToken());
+
+            //Assert
+            Assert.AreEqual(response.EmailShouldBePresent, false);
+        }
+
+        [Test]
+        public async Task WhenCohortIsApprovedBefore9Sept21_ThenEmailShouldBePresentShouldBeFalse()
```

Comment:
Can we please add a test for the black list,
If approved after 9/9 but it is in black list - email present should be false
But pls let me know, if my understanding is incorrect about this.

Reply:
Unfortunately your understanding is slightly wrong here. Providers and Employers can be added to the blacklist for a short period (ie to allow them to approve cohorts which were nearly complete, but not approved on 9-9, or in the case of MOD's current requirements). This means the intent of the field EmailShouldBePresent is not affected by the black list. It is a field which is simply determining 'That the Apprenticeship was created after 9-9-21' and in this case the email should ideally be present.

Perhaps the name is bad?

Reply:
Thanks for the explanation